src/librustc/ty/context.rs

```diff
@@ -496,6 +496,9 @@ pub struct GlobalCtxt<'tcx> {
     /// Cache for layouts computed from types.
     pub layout_cache: RefCell<FnvHashMap<Ty<'tcx>, &'tcx Layout>>,
 
+    //Used to prevent layout from recursing too deeply.
```

Comment:
You can't, because the recursion doesn't go through a dedicated context. You might be able to put it in `TargetDataLayout`, I guess, but that's about it.